drivers/gpio/gpio_altera_pio.c

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (c) 2022, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT altr_pio
+
+/**
+ * @file
+ * @brief PIO driver for Intel FPGA PIO Core IP
+ * Reference : Embedded Peripherals IP User Guide : 27. PIO Core
+ */
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/printk.h>
+#include <inttypes.h>
+#include "gpio_utils.h"
+#include <string.h>
+#include <stdbool.h>
+
+#define ALTERA_AVALON_PIO_DATA_OFFSET		0x00
+#define ALTERA_AVALON_PIO_DIRECTION_OFFSET	0x04
+#define ALTERA_AVALON_PIO_IRQ_OFFSET		0x08
+#define ALTERA_AVALON_PIO_SET_BITS		0x10
+#define ALTERA_AVALON_PIO_CLEAR_BITS		0x14
+
+#define DEV_CFG(_dev) ((struct gpio_altera_config *const)(_dev)->config)
```

Comment:
Please get rid of this macro. Using macros for accessing `dev->config` and `dev->data` is discouraged.